packages/utils-worker/tests/test-perf.ts

```diff
@@ -0,0 +1,310 @@
+
+    for (let i = 0; i < 10000; i++) {
+        result += `const a${i} = (x) => {
+            console.log(x);
+        }
+`;
+    }
+
+    result += `document.getElementById('container').appendChild(document.createElement('svg'));`;
+
+    return {
+        content: result,
+        path,
+        type: ResourceType.JS
+    };
+};
+
+const generateHTML = (resources: Resource[] = []) => {
+    let result = `
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="utf-8">
+        <meta name="viewport" content="width=device-width">
+        <title>Basic Hints Test</title>`;
+
+    const css = resources.filter((res) => {
+        return res.type === ResourceType.CSS;
+    });
+
+    if (css.length) {
+        for (const { path } of css) {
+            result += `<link rel="stylesheet" href="${path}"/>`;
+        }
+    }
+
+    result += `
+    </head>
+    <body>
+        <h1>Basic Hints Test</h1>
+        <dialog open>
+            <form>`;
+
+    for (let i = 0; i < 500; i++) {
+        result += `<img src="icon.png"><button class="button">Test</button>
+                  `;
+    }
+
+    result += `</form>
+        </dialog>
+    </body>
+`;
+
+    const js = resources.filter((res) => {
+        return res.type === ResourceType.JS;
+    });
+
+    if (js.length) {
+        for (const { path } of js) {
+            result += `<script src="${path}"></script>`;
+        }
+    }
+
+    result += `
+</html>
+`;
+
+    return result;
+};
+
+const generateDeepHTML = (resources: Resource[] = []) => {
+    const deep = 100;
+    let result = `
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="utf-8">
+        <meta name="viewport" content="width=device-width">
+        <title>Basic Hints Test</title>`;
+
+    const css = resources.filter((res) => {
+        return res.type === ResourceType.CSS;
+    });
+
+    if (css.length) {
+        for (const { path } of css) {
+            result += `<link rel="stylesheet" href="${path}"/>`;
+        }
+    }
+
+    result += `
+    </head>
+    <body>
+        <h1>Basic Hints Test</h1>
+        <dialog open>
+            <form>`;
+
+    for (let j = 0; j < 5; j++) {
+        for (let i = 0; i < deep; i++) {
+            result += `<div>
+            <img src="icon.png"><button>Test</button>
+            `;
+        }
+
+        for (let i = 0; i < deep; i++) {
+            result += `</div>`;
+        }
+    }
+
+    result += `</form>
+        </dialog>
+    </body>
+`;
+
+    const js = resources.filter((res) => {
+        return res.type === ResourceType.JS;
+    });
+
+    if (js.length) {
+        for (const { path } of js) {
+            result += `<script src="${path}"></script>`;
+        }
+    }
+
+    result += `
+</html>
+`;
+
+    return result;
+};
+
+const prefixOrderCSS = [generateCSS('./index.css')];
+const leadingDotClasslistJS = [generateJS('./index.js')];
+
+const tests: Test[] = [{
+    expectedHints: ['axe/text-alternatives'],
+    expectedTime: 26000,
```

Comment:
Can we set this as close as possible to the real expected time, then fail the test if we're +/- a certain amount (e.g. 10% faster or slower)? My thinking here is we should fail the test if we improve performance so we are reminded to update the values to catch future regressions. We should just make sure the output makes it clear that this was an improvement and the most likely resolution is to update the test.